modin/engines/ray/utils.py

```diff
@@ -84,7 +90,9 @@ def initialize_ray():
                 logging_level=100,
             )
         elif cluster == "":
+            num_cpus = os.environ.get("MODIN_CPUS", None) or multiprocessing.cpu_count()
```

Comment:
Since this was moved, we still need to get the number of CPUs so that `DEFAULT_NPARTITIONS` can be set.

Reply:
this is already done for Ray and Dask, do we need it for Python engine?

Reply:
My question was in the cluster mode, how is `num_cpus` set?

Reply:
If I understand correctly, when we're in the "cluster mode" it is expected that Ray has already been started for us on the head node. In this case we get the `num_cpus` from it.

That is the case for at least remote Ray.

P.S. I think that new enough Ray now even throws an exception when `num_cpus` is set when `ray.init()` is called to connect to an existing cluster.

Reply:
Yes, that is correct, but `DEFAULT_NPARTITIONS` is used to determine internal partitioning. It gets set from `num_cpus`. If we do not set it, we end up with 4 partitions in the whole cluster (the default).

Reply:
I see that `DEFAULT_NPARTITIONS` is correctly updated each time we switch contexts.

Reply:
Do note that each time a context (or, rather, execution engine) is changed, `num_cpus` is computed anew and `DEFAULT_NPARTITIONS` is computed from it afresh as well.

So I think this code behaves as intended.